File: messaging/content-templates/create-whatsapp-card/create-whatsapp-card.java
package Examples;

import com.twilio.rest.content.v1.Content;

import java.util.Arrays;
import java.util.Map;

public class WhatsAppCard {
    public static String CreateTemplate() {
        var waCard = new Content.WhatsappCard();
        waCard.setBody("Congratulations, you have reached Elite status! Add code {{1}} for 10% off.");
        waCard.setHeaderText("This is a {{1}} card");
        waCard.setFooter("To unsubscribe, reply Stop");

        var action1 = new Content.CardAction();
        action1.setType(Content.CardActionType.URL);
        action1.setUrl("https://www.twilio.com");
        action1.setTitle("Order Online");

        var action2 = new Content.CardAction();
        action1.setType(Content.CardActionType.PHONE_NUMBER);
        action1.setPhone("+15551234567");
        action1.setTitle("Call Us");

        waCard.setActions(Arrays.asList(action1));

        var types = new Content.Types();
        types.setWhatsappCard(waCard);

        var createRequest = new Content.ContentCreateRequest("en", types);
        createRequest.setFriendlyName("owl_coupon_code");
        createRequest.setVariables(Map.of(
        "1", "coupon_code"
        ));

        var content = Content.creator(createRequest).create();

        return content.getSid();
    }
}

File: messaging/content-templates/create-whatsapp-card/create-whatsapp-card.cs
// Install the C# / .NET helper library from twilio.com/docs/csharp/install

using System;
using Twilio;
using Twilio.Rest.Content.V1;

     TwilioClient.Init(accountSid, authToken);

    // define the whatsapp/card 
    var whatsappCard = new WhatsappCard.Builder();
    whatsappCard.WithBody("Congratulations, you have reached Elite status! Add code {{1}} for 10% off.");
    whatsappCard.WithHeaderText("This is a {{1}} card");
    whatsappCard.WithFooter("To unsubscribe, reply Stop");
    var cardAction1 = new CardAction.Builder()
        .WithType(CardActionType.Url)
        .WithUrl("https://www.twilio.com")
        .WithTitle("Order Online")
        .Build();
    var cardAction2 = new CardAction.Builder()
        .WithType(CardActionType.PhoneNumber)
        .WithPhone("+15551234567")
        .WithTitle("Call Us")
        .Build();
    whatsappCard.WithActions(new List<CardAction>() { cardAction1, cardAction2 });

    // define all the content types to be part of the template
    var types = new Types.Builder();
    types.WithWhatsappCard(whatsappCard.Build());

    // build the create request object
    var contentCreateRequest = new ContentCreateRequest.Builder();
    contentCreateRequest.WithTypes(types.Build());
    contentCreateRequest.WithLanguage("en");
    contentCreateRequest.WithFriendlyName("owl_coupon_code");
    contentCreateRequest.WithVariables(new Dictionary<string, string>() { {"1", "coupon_code"} });
    
    // create the twilio template
    var contentTemplate = await CreateAsync(contentCreateRequest.Build());

    Console.WriteLine($"Created Twilio Content Template SID: {contentTemplate.Sid}");
